Index: src/test/java/pages/LoginPage.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package pages;\n\nimport org.junit.Assert;\nimport org.openqa.selenium.WebDriver;\nimport org.openqa.selenium.WebElement;\nimport org.openqa.selenium.support.FindBy;\nimport org.openqa.selenium.support.PageFactory;\n\npublic class LoginPage {\n    public LoginPage(WebDriver driver) {\n        PageFactory.initElements(driver, this);\n    }\n\n    @FindBy(id = \"email\")\n    WebElement email;\n\n    @FindBy(id = \"password\")\n    WebElement password;\n\n    @FindBy(xpath = \"//button[@type='submit']\")\n    WebElement loginButton;\n\n    public void login(String email,String password){\n        this.email.sendKeys(email);\n        this.password.sendKeys(password);\n        loginButton.click();\n    }\n\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/pages/LoginPage.java b/src/test/java/pages/LoginPage.java
--- a/src/test/java/pages/LoginPage.java	
+++ b/src/test/java/pages/LoginPage.java	
@@ -8,6 +8,7 @@
 
 public class LoginPage {
     public LoginPage(WebDriver driver) {
+
         PageFactory.initElements(driver, this);
     }
 
Index: src/test/java/stepdefinitions/VerifySubModulesSteps.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/stepdefinitions/VerifySubModulesSteps.java b/src/test/java/stepdefinitions/VerifySubModulesSteps.java
new file mode 100644
--- /dev/null	
+++ b/src/test/java/stepdefinitions/VerifySubModulesSteps.java	
@@ -0,0 +1,49 @@
+package stepdefinitions;
+
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.openqa.selenium.WebDriver;
+import pages.LoginPage;
+import pages.SalesSubmodulesPage;
+import utils.DriverHelper;
+
+import java.util.concurrent.TimeUnit;
+
+public class VerifySubModulesSteps {
+
+    WebDriver driver = DriverHelper.getDriver();
+    LoginPage loginPage = new LoginPage(driver);
+    SalesSubmodulesPage salesSubmodulesPage;
+
+
+    @Given("the employee navigates to the {string}")
+    public void the_employee_navigates_to_the(String url) {
+        driver.get(url);
+        driver.manage().timeouts().implicitlyWait(3, TimeUnit.SECONDS);
+    }
+
+    @When("the employee enter email {string} password {string}")
+    public void the_employee_enter_email_password(String username, String password) {
+        loginPage.login(username, password);
+
+    }
+    @And("the employee clicks on Sales Button in navigation menu from lest side")
+    public void theEmployeeClicksOnSalesButtonInNavigationMenuFromLestSide() {
+        salesSubmodulesPage = new SalesSubmodulesPage(driver);
+        salesSubmodulesPage.clickOnSalesButton();
+
+    }
+    @When("the employee observes the list of available submodules")
+    public void the_employee_observes_the_list_of_available_submodules() {
+        salesSubmodulesPage.getListOfSubmodules();
+
+    }
+    @Then("the employee verifies the correct order of submodules based on their expected sequence")
+    public void the_employee_verifies_the_correct_order_of_submodules_based_on_their_expected_sequence() {
+        salesSubmodulesPage.validateTheRightOrder();
+
+    }
+
+}
Index: src/test/java/pages/SalesSubmodulesPage.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/pages/SalesSubmodulesPage.java b/src/test/java/pages/SalesSubmodulesPage.java
new file mode 100644
--- /dev/null	
+++ b/src/test/java/pages/SalesSubmodulesPage.java	
@@ -0,0 +1,48 @@
+package pages;
+
+import org.junit.Assert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.List;
+
+public class SalesSubmodulesPage {
+
+    private WebDriver driver;
+
+     public SalesSubmodulesPage(WebDriver driver){
+         this.driver = driver;
+         PageFactory.initElements(driver, this);
+     }
+
+     @FindBy(xpath = "//span[contains(text(),'Sales')]")
+     private WebElement salesDropDown;
+
+     @FindBy(xpath = "//ul[contains(@class,'nav nav-second-level collapse in')]//li[contains(@class,'sub-menu-item')]") //ul[contains(@class,'nav nav-second-level collapse in')]
+     private List<WebElement> listOfSubmodules;
+
+     public void clickOnSalesButton() {
+         WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(3));
+         wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//span[contains(text(),'Sales')]")));
+         salesDropDown.click();
+
+     }
+     public void getListOfSubmodules() {
+         int expectedNumberOfSubmodules = 6;
+         Assert.assertEquals("Quantity is not matching", expectedNumberOfSubmodules, listOfSubmodules.size());
+     }
+     public void validateTheRightOrder(){
+         String[] expectedSubmodulesOrder = {"Proposals", "Estimates", "Invoices", "Payments", "Credit Notes", "Items"};
+
+         for (int i = 0; i < expectedSubmodulesOrder.length; i++) {
+             String actualSubmoduleName = listOfSubmodules.get(i).getText();
+             Assert.assertEquals("Submodule order is incorrect", expectedSubmodulesOrder[i], actualSubmoduleName);
+         }
+     }
+}
Index: src/test/resources/features/VerifySubModules.feature
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/resources/features/VerifySubModules.feature b/src/test/resources/features/VerifySubModules.feature
new file mode 100644
--- /dev/null	
+++ b/src/test/resources/features/VerifySubModules.feature	
@@ -0,0 +1,10 @@
+Feature: Validate Submodules of sales and their order
+
+  Background:
+    Given the employee navigates to the "https://techtorialperfex.com/perfexcrm/admin/"
+
+  Scenario: Login with correct employee credentials and add validate order of submodules
+    When the employee enter email "john-techtorial@techtorialperfex.com" password "techtori@l123!"
+    And the employee clicks on Sales Button in navigation menu from lest side
+    And the employee observes the list of available submodules
+    Then the employee verifies the correct order of submodules based on their expected sequence
